ui.go

```diff
@@ -123,6 +128,10 @@ func (u *BasicUi) Output(message string) {
 	fmt.Fprint(u.Writer, "\n")
 }
 
+func (u *BasicUi) Print(message string) {
+	fmt.Fprint(u.Writer, message)
+}
```

Comment:
Can you rename `Print` to `WriteString` and `message` to `msg`?  Also:

```
n, err := io.WriteString(u.Writer, msg); err != nil || len(msg) != n {
  // some kind of error handling
}
```

At the call site we need to make sure we're not using a `bufio`, otherwise we will need to call `Flush()` on `u.Writer`.